pkg/syncer/syncer.go

```diff
@@ -407,7 +411,71 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 
 	// Start tunneler for POD access
 	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
-		go startSyncerTunnel(ctx, upstreamConfig, downstreamConfig, logicalcluster.From(syncTarget), cfg.SyncTargetName)
+		filter := func(http *http.Request) bool {
+			// Allow only requests prefixed by /api/v1/namespaces/.
+			if !strings.HasPrefix(http.URL.Path, "/api/v1/namespaces/") {
+				return false
+			}
+
+			// Ensure that requests are only for pods in a namespace owned by this syncer, if not, return false.
+			nsInformer, err := ddsifForDownstream.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"})
+			if err != nil {
+				logger.V(4).Error(err, "Failed to get informer for namespaces")
+				return false
+			}
+			downstreamNamespaceName := strings.Split(strings.TrimPrefix(http.URL.Path, "/api/v1/namespaces/"), "/")[0]
```

Comment:
Can we use the info resolver please?

Reply:
The request context doesn't seem to contain the required info, creating the http.RequestInfo fails.

Reply:
What is the error?

Reply:
I went with trying to get the request info from the context, and that failed, now after looking at the infoResolver, I've got it working fine! thanks, PTAL